recon/src/recon/sqlitestore.rs

```diff
@@ -312,83 +281,87 @@ where
             key DESC
         LIMIT
             1
-        ; ";
-        let mut stmt = self.conn.prepare(query).unwrap();
-        let rows = stmt
-            .query_map(
-                (
-                    &self.sort_key,
-                    left_fencepost.as_bytes(),
-                    right_fencepost.as_bytes(),
-                ),
-                |row| -> Result<Self::Key, rusqlite::Error> {
-                    Ok(K::from(row.get::<usize, Vec<u8>>(0).unwrap()))
-                },
-            )
-            .unwrap()
-            .map(|row| row.unwrap().clone());
-        let rows = rows.collect::<Vec<Self::Key>>();
-        rows.get(0).cloned()
+        ;",
+        );
+        let rows = query
+            .bind(&self.sort_key)
+            .bind(left_fencepost.as_bytes())
+            .bind(right_fencepost.as_bytes())
+            .fetch_all(&self.pool)
+            .await?;
+        Ok(rows.get(0).map(|row| {
+            let bytes: Vec<u8> = row.get(0);
+            K::from(bytes)
+        }))
     }
 
     #[instrument(skip(self))]
-    fn first_and_last(
-        &self,
+    async fn first_and_last(
```

Comment:
todo
```
SELECT first.key, last.key
FROM 
	(
		SELECT key
		FROM recon
		WHERE
            		sort_key = ? AND
            		key > ? AND key < ?
		ORDER BY cid ASC
		LIMIT 1
	) as first
JOIN
	(
		SELECT key
		FROM recon
		WHERE
            		sort_key = ? AND
            		key > ? AND key < ?
		ORDER BY cid DESC
		LIMIT 1
	) as last
```

Reply:
I'll submit a follow up PR for this change.